utils/rowgroup/rowaggregation.cpp

```diff
@@ -4560,7 +4572,8 @@ void RowAggregationDistinct::addRowGroup(const RowGroup* pRows,
 //------------------------------------------------------------------------------
 void RowAggregationDistinct::doDistinctAggregation()
 {
-  while (dynamic_cast<RowAggregationUM*>(fAggregator.get())->nextRowGroup())
+  fAggregator->finalAggregation();  // TODO: Perhaps move somewhere else?
+  while (dynamic_cast<RowAggregationUM*>(fAggregator.get())->nextOutputRowGroup())
```

Comment:
Every dynamic_cast costs a lot, so I suggest to cast it only once. I also suggest to check dynamic_cast result before the cast is used.